### PR TITLE
Allow any user with egghead account to view "account management" page

### PR DESCRIPTION
**Pull request by jbranchaud**
The account management page has been restricted to users with active subscriptions. That was necessitated by the slug in the URL which corresponded to their membership. This PR moves the account management page from `/accounts/<slug>` to `/user`. It can. now be visited by anyone. Only users with an active membership will see their subscription details.

### CODE REVIEW

File: src/components/app/header.tsx [hunk 1]
```diff
@@ -69,35 +69,14 @@ const Header: FunctionComponent = () => {
                 </a>
               </Link>
             )}
-            {!isEmpty(viewer.accounts) ? (
```
Comment (jbranchaud) [inline src/components/app/header.tsx:72]: Remove conditional

File: src/components/app/header.tsx [hunk 2]
```diff
@@ -69,35 +69,14 @@ const Header: FunctionComponent = () => {
...
-                    {viewer.is_pro && ' ⭐️'}
-                  </span>
-                </a>
-              </Link>
-            ) : (
-              <div
-                className="flex items-center space-x-2 p-3"
-                onClick={() => {
-                  track(`clicked user name area`, {
+            <Link href="/user">
```
Comment (jbranchaud) [inline src/components/app/header.tsx:72]: Link to `/user`

File: src/components/users/request-email-change-form.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1/email_change_requests`,
+    {
+      email: newEmail,
+    },
+  )
+
+  return data
+}
+
+const RequestEmailChangeForm: React.FunctionComponent<RequestEmailChangeFormProps> = ({
```
Comment (jbranchaud) [inline src/components/users/request-email-change-form.tsx:25]: Move this to its own file to de-clutter the `user/index.tsx` page file.

File: src/pages/api/stripe/billing/session.ts [hunk 4]
```diff
@@ -13,13 +13,9 @@ const StripeCheckoutSession = async (
...
       if (!customer_id) throw new Error('no customer id')
 
-      const account_slug = req.query.account_slug
-
       const session = await stripe.billingPortal.sessions.create({
         customer: customer_id,
-        return_url: account_slug
-          ? `${process.env.NEXT_PUBLIC_DEPLOYMENT_URL}/accounts/${account_slug}`
-          : process.env.NEXT_PUBLIC_DEPLOYMENT_URL,
+        return_url: `${process.env.NEXT_PUBLIC_DEPLOYMENT_URL}/user`,
```
Comment (jbranchaud) [inline src/pages/api/stripe/billing/session.ts:18]: This can link directly to `/user` now

File: src/pages/email-change-request/[token].tsx [hunk 5]
```diff
@@ -40,11 +40,7 @@ const EmailChangeRequest: React.FunctionComponent = () => {
                       // refresh the user to get the latest email before redirecting
                       await refreshUser()
 
-                      if (isEmpty(viewer.accounts)) {
-                        router.replace('/')
-                      } else {
-                        router.replace(`/accounts/${viewer.accounts[0].slug}`)
-                      }
+                      router.replace('/user')
```
Comment (jbranchaud) [inline src/pages/email-change-request/[token].tsx:43]: Same, this too can link directly to `/user`

File: src/pages/user/index.tsx [hunk 6]
```diff
@@ -7,157 +7,24 @@ import axios from 'axios'
...
-    `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1/email_change_requests`,
-    {
-      email: newEmail,
-    },
-  )
-
-  return data
-}
-
-const RequestEmailChangeForm: React.FunctionComponent<RequestEmailChangeFormProps> = ({
```
Comment (jbranchaud) [inline src/pages/user/index.tsx:32]: This all lives in another file now 🎉

File: src/pages/user/index.tsx [hunk 7]
```diff
@@ -7,157 +7,24 @@ import axios from 'axios'
...
-  )
-  const account: any =
-    params?.slug && (await loadAccount(params.slug as string, eggheadToken))
-  return {
-    props: {
-      loginRequired,
-      account,
-    },
-  }
-}
```
Comment (jbranchaud) [inline src/pages/user/index.tsx:146]: This logic gets moved to a `useEffect`

File: src/pages/user/index.tsx [hunk 8]
```diff
@@ -169,6 +36,18 @@ const Account: React.FunctionComponent<
...
+  React.useEffect(() => {
+    const loadAccountForSlug = async (slug) => {
+      if (slug) {
+        const account: any = await loadAccount(slug, authToken)
+        setAccount(account)
+      }
+    }
+
+    loadAccountForSlug(slug)
+  }, [slug, authToken])
```
Comment (jbranchaud) [inline src/pages/user/index.tsx:49]: Get the slug from `useViewer()` above and then `loadAccount` with that. Once the account is loaded, if there is a `stripe_customer_id`, then the other useEffect will be able to look up that info.

**Review by joelhooks — APPROVED** [review]

**Pull request merged**